src/UsgsAstroProjectedLsSensorModel.cpp

```diff
@@ -0,0 +1,799 @@
+  reset();
+
+  auto j = stateAsJson(stateString);
+  m_geoTransform = j["m_geoTransform"].get<std::vector<double>>();
+  m_projString = j["m_projString"];
+  MESSAGE_LOG(
+      spdlog::level::trace,
+      "m_geoTransform: {} "
+      "m_projString: {} ",
+      j["m_geoTransform"].dump(), j["m_projString"].dump());
+  UsgsAstroLsSensorModel::replaceModelState(stateString);
+}
+
+//***************************************************************************
+// UsgsAstroLineScannerSensorModel::getModelNameFromModelState
+//***************************************************************************
+std::string UsgsAstroProjectedLsSensorModel::getModelNameFromModelState(
+    const std::string& model_state) {
+  // Parse the string to JSON
+  auto j = stateAsJson(model_state);
+  // If model name cannot be determined, return a blank string
+  std::string model_name;
+
+  if (j.find("m_modelName") != j.end()) {
+    model_name = j["m_modelName"];
+  } else {
+    csm::Error::ErrorType aErrorType = csm::Error::INVALID_SENSOR_MODEL_STATE;
+    std::string aMessage = "No 'm_modelName' key in the model state object.";
+    std::string aFunction = "UsgsAstroProjectedLsPlugin::getModelNameFromModelState";
+    csm::Error csmErr(aErrorType, aMessage, aFunction);
+    throw(csmErr);
+  }
+  if (model_name != _SENSOR_MODEL_NAME) {
+    csm::Error::ErrorType aErrorType = csm::Error::SENSOR_MODEL_NOT_SUPPORTED;
+    std::string aMessage = "Sensor model not supported.";
+    std::string aFunction = "UsgsAstroProjectedLsPlugin::getModelNameFromModelState()";
+    csm::Error csmErr(aErrorType, aMessage, aFunction);
+    throw(csmErr);
+  }
+  return model_name;
+}
+
+//***************************************************************************
+// UsgsAstroLineScannerSensorModel::getModelState
+//***************************************************************************
+std::string UsgsAstroProjectedLsSensorModel::getModelState() const {
+  auto state = stateAsJson(UsgsAstroLsSensorModel::getModelState());
+  state["m_geoTransform"] = m_geoTransform;
+  state["m_projString"] = m_projString;
+  MESSAGE_LOG(
+      spdlog::level::trace,
+      "m_geoTransform: {}, {}, {}, {}, {}, {} "
+      "m_projString: {} ",
+      m_geoTransform[0], 
+      m_geoTransform[1], 
+      m_geoTransform[2], 
+      m_geoTransform[3], 
+      m_geoTransform[4], 
+      m_geoTransform[5], 
+      m_projString);
+  // Use dump(2) to avoid creating the model string as a single long line
+  std::string stateString = getModelName() + "\n" + state.dump(2);
+  return stateString;
+}
+
+//***************************************************************************
+// UsgsAstroLineScannerSensorModel::reset
+//***************************************************************************
+void UsgsAstroProjectedLsSensorModel::reset() {
+  MESSAGE_LOG(spdlog::level::debug, "Running reset()");
+
+  m_geoTransform = std::vector<double>(6, 0.0);
+  m_projString = "";
+}
```

Comment:
Should we call the parent reset?

Reply:
The parent class calls its reset in `UsgsAstroLsSensorModel::replaceModelState`, so we don't need to call it again